cmd/relay-gossip/main.go

```diff
@@ -99,46 +100,156 @@ var runCmd = &cli.Command{
-			return err
+
+		// Try and build a default gossip relay using other options that might have been set.
+		if len(hashesMap) == 0 &&
+			len(groupConfs) == 0 &&
+			!hashFlagSet {
+			if err := boostrapGossipRelayNode(cctx, ""); err != nil {
+				return err
+			}
 		}
-		<-chan int(nil)
-		return nil
+
+		// Wait indefinitely for our client(s) to run
+		<-cctx.Context.Done()
+
+		return cctx.Context.Err()
 	},
 }
 
+func computeGroupConfs(cctx *cli.Context) []string {
+	if !cctx.IsSet(lib.GroupConfFlag.Name) {
+		return nil
+	}
+
+	groupConfs := cctx.String(lib.GroupConfFlag.Name)
+	return strings.Split(groupConfs, ",")
+}
+
+func boostrapGossipRelayNode(cctx *cli.Context, chainHash string) error {
+	c, err := lib.Create(cctx, cctx.IsSet(metricsFlag.Name))
+	if err != nil {
+		return fmt.Errorf("constructing client: %w", err)
+	}
+
+	chainInfo, err := c.Info(cctx.Context)
+	if err != nil {
+		return fmt.Errorf("cannot retrieve chain info: %w", err)
+	}
+
+	// Set the path to be desired 'storage path / beaconID'.
+	// This allows running multiple networks via the same beacon.
+	dataDir := path.Join(cctx.String(storeFlag.Name), chainInfo.ID)
+
+	cfg := &lp2p.GossipRelayConfig{
+		ChainHash:    chainHash,
+		BeaconID:     chainInfo.ID,
+		PeerWith:     cctx.StringSlice(peerWithFlag.Name),
+		Addr:         cctx.String(listenFlag.Name),
+		DataDir:      dataDir,
+		IdentityPath: cctx.String(idFlag.Name),
+		Client:       c,
+	}
+	_, err = lp2p.NewGossipRelayNode(log.DefaultLogger(), cfg)
+	if err != nil {
+		err = fmt.Errorf("could not initialize a new gossip relay node %w", err)
+	}
+	return err
+}
+
+func computeHashesMap(hashMaps, groupConfs []string) (map[string]string, error) {
+	hashesMap := make(map[string]string)
+	if len(hashMaps) == 0 {
+		//nolint:nilnil // We want to return an nil instead of an empty map.
+		return nil, nil
+	}
+
+	if groupConfs != nil {
+		// Here we overload the -group-conf flag and we convert it from String to StringSlice.
+		// This is required to provide the correct -group-conf value for the client library.
+		if len(groupConfs) > 0 &&
+			len(groupConfs) != len(hashMaps) {
+			return nil, fmt.Errorf("list of provided hashes is different from list of group-conf files")
+		}
+	} else {
+		for i := 0; i < len(hashMaps); i++ {
+			groupConfs = append(groupConfs, "")
+		}
+	}
+
+	for idx, hash := range hashMaps {
+		if _, err := hex.DecodeString(hash); err != nil {
+			return nil, fmt.Errorf("decoding hash: %w", err)
+		}
+		hashesMap[hash] = groupConfs[idx]
+	}
+
+	return hashesMap, nil
+}
+
 var clientCmd = &cli.Command{
 	Name:  "client",
 	Flags: lib.ClientFlags,
 	Action: func(cctx *cli.Context) error {
 		c, err := lib.Create(cctx, false)
 		if err != nil {
-			return xerrors.Errorf("constructing client: %w", err)
+			return fmt.Errorf("constructing client: %w", err)
 		}
 
-		for rand := range c.Watch(context.Background()) {
+		for rand := range c.Watch(cctx.Context) {
```

Comment:
cctx is not the idiomatic name for the context variable. I'm guessing this is old code?

Reply:
It's old code. `cctx`, I think, comes from the `commandCtx`, as `cctx` is not a `context.Context` type but rather the CLI library's Context.